contrib/pg_stat_statements/sql/test_pg_stat_statements.sql

```diff
@@ -0,0 +1,215 @@
+-- start_ignore
+\! gpconfig -c shared_preload_libraries -v 'pg_stat_statements';
+\! gpstop -raq -M fast;
+\c
+CREATE EXTENSION IF NOT EXISTS pg_stat_statements;
+DROP TABLE IF EXISTS table_test_pg_stat_statements;
+-- end_ignore
+-- simple test to check there is no warnings during jumbling of the query
+CREATE TABLE table_test_pg_stat_statements
+(
+    item1 integer,
+    item2 integer,
+    item3 integer,
+    item4 integer
+)
+DISTRIBUTED BY (item1);
+
+SELECT   
+    SUM(item4) AS sum_item1
+   ,item2
+   ,GROUPING(item1)+GROUPING(item2) AS item_1_2_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
+
+-- start tests for pg_stat_statements logic
+
+-- Known issue: query is not added to pg_stat_statements statistics in
+-- case it is planned by GPORCA. So disable GPORCA during tests.
+SET optimizer='off';
+
+SELECT pg_stat_statements_reset();
+
+-- launch 2 equivalent queries  
+SELECT
+    GROUPING(item1) AS item_1_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
+
+SELECT
+    GROUPING(item1) AS item_1_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 1
+GROUP BY ROLLUP(item1, item2);
+
+-- pg_stat_statements statistics should have 2 calls for 1 entry
+-- corresponding to the two preceding SELECT queries
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- launch not equivalent query
+SELECT
+    GROUPING(item2) AS item_1_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
+
+-- check that it has separate entry
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- change alias name
+SELECT
+    GROUPING(item2) AS item_2_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
```

Comment:
changing alias will not affect queryid even without the patch. therefore, there is no point in testing it.

Reply:
will remove this part